src/solver/variable/adequacy/overallCost.h

```diff
@@ -251,39 +251,33 @@ class OverallCost : public Variable::IVariable<OverallCost<NextT>, NextT, VCardO
 
     void hourForEachArea(State& state, unsigned int numSpace)
     {
+        auto area = state.area;
         // Total UnsupliedEnergy emissions
         pValuesForTheCurrentYear[numSpace][state.hourInTheYear] +=
           // Current Hydro Storage generation
           (state.hourlyResults->ValeursHorairesDeDefaillancePositive[state.hourInTheWeek]
-           * state.area->thermal.unsuppliedEnergyCost)
+           * area->thermal.unsuppliedEnergyCost)
           + ((state.hourlyResults->ValeursHorairesDeDefaillanceNegative[state.hourInTheWeek]
-              + state.resSpilled.entry[state.area->index][state.hourInTheWeek])
-             * state.area->thermal.spilledEnergyCost);
+              + state.resSpilled.entry[area->index][state.hourInTheWeek])
+             * area->thermal.spilledEnergyCost);
 
         // Hydro costs : water value and pumping
         pValuesForTheCurrentYear[numSpace].hour[state.hourInTheYear]
           += state.problemeHebdo->CaracteristiquesHydrauliques[state.area->index]
                ->WeeklyWaterValueStateRegular
              * (state.hourlyResults->TurbinageHoraire[state.hourInTheWeek]
-                - state.area->hydro.pumpingEfficiency
+                - area->hydro.pumpingEfficiency
                     * state.hourlyResults->PompageHoraire[state.hourInTheWeek]);
 
-        // Next variable
-        NextType::hourForEachArea(state, numSpace);
-    }
-
-    void hourForClusters(State& state, unsigned int numSpace)
-    {
-        // Total OverallCost
-        for (uint clusterIndex = 0; clusterIndex != state.area->thermal.clusterCount();
-             ++clusterIndex)
+        // Thermal costs
+        for (uint clusterIndex = 0; clusterIndex != area->thermal.clusterCount(); ++clusterIndex)
         {
             pValuesForTheCurrentYear[numSpace][state.hourInTheYear]
-              += state.thermalClustersOperatingCost[state.area->index][clusterIndex];
+              += state.thermal[area->index].thermalClustersOperatingCost[clusterIndex];
```

Comment:
You define `auto area = state.area;` above, why not defining **thermal** in the same way ?
This remark applies to all output variable classes involving clusters (of all kind).